### PR TITLE
morogrim tidewalker https://github.com/Looking4Group/L4G_Core/issues/2733

### PR DESCRIPTION
**Pull request by tst12**

### CODE REVIEW

File: src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp [hunk 1]
```diff
@@ -39,7 +39,6 @@ EndScriptData */
 #define EMOTE_EARTHQUAKE            -1548040
 #define EMOTE_WATERY_GLOBULES       -1548041
 
-#define SPELL_THRASH_PASSIVE        19818
```
Comment (AnonXS) [inline src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp]: https://github.com/cmangos/mangos-tbc/blob/master/src/scriptdev2/scripts/outland/coilfang_reservoir/serpent_shrine/boss_morogrim_tidewalker.cpp#L42

File: src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp [hunk 2]
```diff
@@ -103,13 +102,12 @@ struct boss_morogrim_tidewalkerAI : public ScriptedAI
     void Reset()
     {
         TidalWave_Timer = 10000;
-        WateryGrave_Timer = 20000;
-        Earthquake_Timer = 45000;
+        WateryGrave_Timer = 30000;
```
Comment (AnonXS) [inline src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp]: https://github.com/cmangos/mangos-tbc/blob/master/src/scriptdev2/scripts/outland/coilfang_reservoir/serpent_shrine/boss_morogrim_tidewalker.cpp#L94 ff.

File: src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp [hunk 3]
```diff
@@ -103,13 +102,12 @@ struct boss_morogrim_tidewalkerAI : public ScriptedAI
...
+        Earthquake_Timer = 40000;
         WateryGlobules_Timer = 0;
 
         Earthquake = false;
         Phase2 = false;
-        m_creature->CastSpell(m_creature, SPELL_THRASH_PASSIVE, true);
```
Comment (AnonXS) [inline src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp]: https://github.com/cmangos/mangos-tbc/blob/master/src/scriptdev2/scripts/outland/coilfang_reservoir/serpent_shrine/boss_morogrim_tidewalker.cpp#L101

File: src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp [hunk 4]
```diff
@@ -192,12 +191,11 @@ struct boss_morogrim_tidewalkerAI : public ScriptedAI
...
                 for (std::list<Unit*>::const_iterator itr = tmpList.begin(); itr != tmpList.end(); ++itr)
-                {
-                    (*itr)->CastSpell(*itr, wateryGraves[i++], true);
-                }
-                             
+                    (*itr)->CastSpell((*itr), wateryGraves[i++], false);
+
+                DoScriptText(RAND(SAY_SUMMON_BUBL1, SAY_SUMMON_BUBL2), m_creature);
                 DoScriptText(EMOTE_WATERY_GRAVE, m_creature);
-                WateryGrave_Timer = 28500;
```
Comment (AnonXS) [inline src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp]: https://github.com/cmangos/mangos-tbc/blob/master/src/scriptdev2/scripts/outland/coilfang_reservoir/serpent_shrine/boss_morogrim_tidewalker.cpp#L208

File: src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp [hunk 5]
```diff
@@ -164,9 +162,10 @@ struct boss_morogrim_tidewalkerAI : public ScriptedAI
                         Murloc->setActive(true);
                         Murloc->AI()->AttackStart(target);
                     }
-                }                
+                }
+
                 Earthquake = false;
-                Earthquake_Timer = urand(35000, 50000);
```
Comment (AnonXS) [inline src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp]: https://github.com/cmangos/mangos-tbc/blob/master/src/scriptdev2/scripts/outland/coilfang_reservoir/serpent_shrine/boss_morogrim_tidewalker.cpp#L186

File: src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp [hunk 6]
```diff
@@ -216,13 +214,10 @@ struct boss_morogrim_tidewalkerAI : public ScriptedAI
 
                 int i = 0;
                 for (std::list<Unit*>::const_iterator itr = tmpList.begin(); itr != tmpList.end(); ++itr)
-                 {                   
-                    (*itr)->CastSpell((*itr), summonGlobules[i], true);                    
-                    i++;
-                }
-                DoScriptText(EMOTE_WATERY_GLOBULES, m_creature);              
-                DoScriptText(RAND(SAY_SUMMON_BUBL1, SAY_SUMMON_BUBL2), m_creature);                
-                WateryGlobules_Timer = 40000;
```
Comment (AnonXS) [inline src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp]: https://github.com/cmangos/mangos-tbc/blob/master/src/scriptdev2/scripts/outland/coilfang_reservoir/serpent_shrine/boss_morogrim_tidewalker.cpp#L232

**Comment by tst12** [comment]
l4g:
```
Id: 19818
Name: Double Attack
Effect 0:
Id: 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = 1
Targets (1, 0) (TARGET_SELF, NO_TARGET)
Aura Id 42 (SPELL_AURA_PROC_TRIGGER_SPELL), value = 1, misc = 0 (0), miscB = 0, periodic = 0
Trigger spell: 18941 Not found
```

```
//     else if (auraSpellInfo->Id==19818) // Double Attack
//          trigger_spell_id = ;

```
cmangos:
```
Id: 18943
Name: Double Attack
Description: Gives the caster $s1 extra $lattack:attacks;.
Effect 0:
Id: 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = 1
Targets (1, 0) (TARGET_SELF, NO_TARGET)
Aura Id 42 (SPELL_AURA_PROC_TRIGGER_SPELL), value = 1, misc = 0 (0), miscB = 0, periodic = 0
Trigger spell: 18941 Not found
```

`                // case 18943: break;                   // Double Attack`

dont worry about things that dont do anything

**Comment by Pbeitelmann** [comment]
```19818 (L4G)

Proc flag 0x00000014, chance = 25```

```18943 cMangos

Proc flag 0x00000014, chance = 50```

We chose 19818 on purpose, since 50% procc chance is way too much and makes the encounter too spikey. We also based this on feedback we received on the T5 PTR.

**Comment by AnonXS** [comment]
As all Double Attack Spells 

http://db.hellfire-tbc.com/?spell=18943
http://db.hellfire-tbc.com/?spell=19194
http://db.hellfire-tbc.com/?spell=19817
http://db.hellfire-tbc.com/?spell=19818 
seem to be nonfunctional due to missing spell proc trigger spell scripting, we might want to use a low % thrash spell like or link them to http://db.hellfire-tbc.com/?spell=21919

http://db.hellfire-tbc.com/?spell=3417 10%
http://db.hellfire-tbc.com/?spell=8876 15%
http://db.hellfire-tbc.com/?spell=12787 35%

File: src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp (hunk 1 above)
Comment (AnonXS) [inline src/scripts/scripts/zone/coilfang_resevoir/serpent_shrine/boss_morogrim_tidewalker.cpp]: #define SPELL_THRASH_PASSIVE        3417

**Comment by Xadras** [comment]
What is the current status of this? :>

**Comment by AnonXS** [comment]
I am unhappy with a solution without doubleattack or trash so we need to come up with an agreement here, it also would be nice to hear what this fixes as there is no official issue for this. so it would be nice to hear how this version works without testing it ourselves.

**Pull request merged**